packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -1873,6 +1873,16 @@ export class Bucket extends BucketBase {
       throw new Error("Cannot use 'noncurrent' rules on a nonversioned bucket");
     }
 
+    rule.transitions?.forEach((transition) => {
+      if (transition.transitionAfter === undefined && transition.transitionDate === undefined) {
+        throw new Error('Exactly one of transitionDate and transitionAfter must be specified');
```

Comment:
If there are two checks, please add customized messaging on what has gone wrong.

Reply:
I combined the two checks into one to match the real error message.